android/src/main/java/com/amplitude/android/internal/fragments/FragmentActivityHandler.kt

```diff
@@ -0,0 +1,32 @@
+package com.amplitude.android.internal.fragments
+
+import android.app.Activity
+import androidx.fragment.app.FragmentActivity
+import com.amplitude.common.Logger
+import java.util.WeakHashMap
+
+internal typealias TrackEventCallback = (String, Map<String, Any?>) -> Unit
+
+internal object FragmentActivityHandler {
+    private val callbacksMap =
+        WeakHashMap<FragmentActivity, AutocaptureFragmentLifecycleCallbacks>()
```

Comment:
This approach will not work if there are multiple instances of Amplitude (the last callback will overwrite the previous one).